xwiki-rendering-macros/xwiki-rendering-macro-html/src/main/java/org/xwiki/rendering/internal/macro/html/HTMLMacro.java

```diff
@@ -152,83 +139,28 @@ public List<Block> execute(HTMLMacroParameters parameters, String content, Macro
                 normalizedContent = renderWikiSyntax(normalizedContent, context.getTransformation(), context);
             }
 
-            // Clean the HTML into valid XHTML if the user has asked (it's the default).
-            if (parameters.getClean()) {
-                normalizedContent = cleanHTML(normalizedContent, context);
-            } else if (context.getTransformationContext().isRestricted()) {
-                throw new MacroExecutionException(
-                    "The HTML macro may not be used with clean=\"false\" in this context.");
+            RawBlock contentBlock = new RawBlock(normalizedContent, targetSyntax);
+
+            try {
+                RawBlockFilterParameters filterParameters = new RawBlockFilterParameters();
+                filterParameters.setClean(parameters.getClean());
+                filterParameters.setMacroTransformationContext(context);
+                for (RawBlockFilter filter
+                    : this.componentManager.<RawBlockFilter>getInstanceList(RawBlockFilter.class)) {
+                    contentBlock = filter.filter(contentBlock, filterParameters);
```

Comment:
So it means each filter will perform a filtering on a block that was already filtered previously. So looks like we should introduce a mechanism to order the filters. For example, a filter could transform some attribute that could be removed by another one if they're not transformed.